elide-graphql/src/main/java/com/yahoo/elide/graphql/ModelBuilder.java

```diff
@@ -150,12 +152,63 @@ public ModelBuilder(EntityDictionary entityDictionary,
         queryObjectRegistry = new HashMap<>();
         connectionObjectRegistry = new HashMap<>();
         excludedEntities = new HashSet<>();
+        relationshipOpArgument = new HashMap<>();
     }
 
     public void withExcludedEntities(Set<Type<?>> excludedEntities) {
         this.excludedEntities = excludedEntities;
     }
 
+    public GraphQLArgument getRelationshipOp(Type<?> clazz) {
+        GraphQLArgument existing = relationshipOpArgument.get(clazz);
+        if (existing != null) {
+            return existing;
+        }
+
+        String entityName = entityDictionary.getJsonAliasFor(clazz);
+        String postfix = entityName.substring(0, 1).toUpperCase(Locale.ENGLISH) + entityName.substring(1);
+        GraphQLEnumType relationshipOp = generator.buildClassToEnumType(ClassType.of(RelationshipOp.class),
+                name -> name + postfix, e -> {
+                    RelationshipOp op = RelationshipOp.valueOf(e.name());
+                    switch (op) {
+                    case FETCH:
+                        return canRead(clazz);
+                    case DELETE:
+                        return canDelete(clazz);
```

Comment:
This would require UpdatePermission - not Delete.

Reply:
Changed to UpdatePermission